Update versions and add more advanced content (#30)
actions-workshop/actions-workshop: docs/002-basics-of-ci-with-actions.md

    strategy:
      matrix:

>         node-version: [18.x, 20.x, 22.x]

Reviewer: Node Version 18 is EOL, and 20 is also almost gone. So while we update, why not just update to the supported version 22 and 24? 🙂 

```suggestion
        node-version: [22.x, 24.x]
```

        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/

    steps:

>       - uses: actions/checkout@v4

Reviewer: V5 is out since a short while 🙂 

```suggestion
      - uses: actions/checkout@v5
```

      - name: Use Node.js ${{ matrix.node-version }}

>         uses: actions/setup-node@v4

Reviewer: ```suggestion
        uses: actions/setup-node@v5
```

- `actions/` references the owner of the action, which is translated into a user or organization on GitHub.
- `setup-node` refers to the name of the action, which corresponds to a repository on GitHub.

> - `@v4` represents the version of the action, which corresponds to a Git tag or a general reference (such as a branch or even a commit SHA) on the repository.

Reviewer: ```suggestion
- `@v5` represents the version of the action, which corresponds to a Git tag or a general reference (such as a branch or even a commit SHA) on the repository.
```

This reference structure makes it straightforward to navigate to the source code of any action by merely appending the `owner` and `name` to the `github.com` URL, like so: `https://github.com/{owner}/{name}`. For the above example, this would be <https://github.com/actions/setup-node>.

### 2.3 - Understanding matrix builds

> Observe that our workflow employs a [matrix build strategy](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) with three Node.js versions: 18, 20, and 22. A matrix build enables you to execute a job in parallel using various input parameters. In our case, we're running the same job twice, but with distinct Node.js versions.

Reviewer: ```suggestion
Observe that our workflow employs a [matrix build strategy](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) with two Node.js versions: 22 and 24. A matrix build enables you to execute a job in parallel using various input parameters. In our case, we're running the same job twice, but with distinct Node.js versions.
```

      contents: read
      pull-requests: write
    steps:

>       - uses: actions/checkout@v4

Reviewer: ```suggestion
      - uses: actions/checkout@v5
```

      - name: Use Node.js

>         uses: actions/setup-node@v4

Reviewer: ```suggestion
        uses: actions/setup-node@v5
```
